### PR TITLE
chore: update workflows and dependabot

### PR DESCRIPTION
**Pull request by nigelhanlon**
This PR makes changes to the CI workflow and updates Dependabot to version 2.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -15,16 +15,20 @@ jobs:
...
-      #     token: ${{ secrets.CODECOV_TOKEN }}
+
+  automerge:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: fastify/github-action-merge-dependabot@v1
+        if: ${{ github.actor == 'dependabot[bot]' && github.event_name == 'pull_request' }}
+        with:
+          github-token: ${{secrets.github_token}}
```
Comment (nigelhanlon) [inline .github/workflows/ci.yml:34]: @simoneb I'm not sure if that secret exists as I don't have access to the project settings.

Comment (simoneb) [reply .github/workflows/ci.yml:34]: the repo you got this from has an issue I guess. see https://docs.github.com/en/actions/reference/authentication-in-a-workflow

Comment (nigelhanlon) [reply .github/workflows/ci.yml]: Ahh, I see so `${{ secrets.GITHUB_TOKEN }}` is created automatically.

Comment (simoneb) [reply .github/workflows/ci.yml]: more specifically, it is always available in a gh action run

File: .github/.dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+version: 2
+updates:
+  - package-ecosystem: npm
+    directory: '/'
+    schedule:
+      interval: daily
+    open-pull-requests-limit: 10
```
Comment (simoneb) [inline .github/.dependabot.yml:7]: please add update of gh actions too

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -15,16 +15,20 @@ jobs:
...
-      #     token: ${{ secrets.CODECOV_TOKEN }}
+
+  automerge:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: fastify/github-action-merge-dependabot@v1
+        if: ${{ github.actor == 'dependabot[bot]' && github.event_name == 'pull_request' }}
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (simoneb) [inline .github/workflows/ci.yml:34]: can you please also fix the repo you got the wrong version of this from? I believe there's a card for it https://github.com/orgs/nearform/projects/3#card-55207086

Comment (nigelhanlon) [reply .github/workflows/ci.yml:34]: Sure, I will update that as well 👍

Comment (nigelhanlon) [reply .github/workflows/ci.yml:34]: Fixed here: https://github.com/nearform/fastify-secrets-azure/pull/28

File: .github/.dependabot.yml (hunk 2 above)
Comment (nigelhanlon) [reply .github/.dependabot.yml:7]: added daily check

**Review by simoneb — APPROVED** [review]

**Pull request merged**